borg/archiver.py

```diff
@@ -121,6 +121,18 @@ def print_file_status(self, status, path):
         if self.output_list and (self.output_filter is None or status in self.output_filter):
             logger.info("%1s %s", status, remove_surrogates(path))
 
+    @staticmethod
+    def build_matcher(inclexcl_patterns, paths):
+        matcher = PatternMatcher()
+        if inclexcl_patterns:
+            matcher.add_inclexcl(inclexcl_patterns)
```

Comment:
One thing that's not entirely clear to me here is how/where RootPaths are segregated when given on the command line.

Suppose: `borg create --pattern "R /foo"`, wouldn't that result in an empty args.paths?